code/game/machinery/bots/screambot.dm

```diff
@@ -0,0 +1,147 @@
+		qdel(src)
+
+/obj/machinery/bot/screambot
+	name = "screambot"
+	desc = "SCREAMING INTENSIFIES."
+	icon = 'icons/obj/aibots_new.dmi'
+	icon_state = "screambot"
+	layer = 5.0
+	density = 0
+	anchored = 0
+	health = 25
+	maxhealth = 25
+	var/cooldown = 0
+	var/speedup = 1
+	var/probability = 30
+	var/list/sounds = list('sound/misc/scream_f1.ogg', 'sound/misc/scream_f2.ogg', 'sound/misc/scream_m1.ogg', 'sound/misc/scream_m2.ogg', 'sound/voice/screamsilicon.ogg', 'sound/misc/cat.ogg', 'sound/misc/caw.ogg', 'sound/misc/lizard.ogg')
+
+/obj/machinery/bot/screambot/explode()
+	visible_message("<span class='userdanger'>[src] blows apart!</span>")
+	var/turf/T = get_turf(src)
+	if(prob(50))
+		new /obj/item/robot_parts/l_arm(T)
+
+	var/datum/effect_system/spark_spread/s = new
+	s.set_up(3, 1, src)
+	s.start()
+
+	new /obj/effect/decal/cleanable/oil(loc)
+	qdel(src)
+
+/obj/machinery/bot/screambot/New()
+	..()
+	icon_state = "screambot[on]"
+
+/obj/machinery/bot/screambot/turn_on()
+	..()
+	icon_state = "screambot[on]"
+
+/obj/machinery/bot/screambot/turn_off()
+	..()
+	icon_state = "screambot[on]"
+
+/obj/machinery/bot/screambot/bot_process()
+	if (!..())
+		return
+
+	if(isturf(src.loc))
+		var/anydir = pick(cardinal)
+		if(Process_Spacemove(anydir))
+			Move(get_step(src, anydir), anydir)
+
+	if(cooldown < world.time && prob(probability)) //Probability so it's not TOO annoying (-atlantique except when its emagged lmao)
+		cooldown = world.time + 100 / speedup
+		if(sounds.len)
+			playsound(loc, pick(sounds), 50, 1, 7, 1.2)
+		flick("screambot_scream", src)
+		visible_message("<span class='danger'><b>[src]</b> screams!</span>")
+
+/obj/machinery/bot/screambot/emag_act(mob/user as mob)
+	if(!emagged)
+		emagged = 1
+		speedup = 10
+		probability = 100
+		user << "<span class='warning'>Nice. The screambot is going to be really, REALLY annoying now.</span>"
+
+//MENU
+
+/obj/machinery/bot/screambot/attack_hand(mob/user as mob)
+	var/dat = "<div class='statusDisplay'>"
+	dat += "Human scream: <A href='?src=\ref[src];action=toggle;scream=human'>[('sound/misc/scream_f1.ogg' in sounds) ? "On" : "Off"]</A><BR>"
+	dat += "Synthesized scream: <A href='?src=\ref[src];action=toggle;scream=silicon'>[('sound/voice/screamsilicon.ogg' in sounds) ? "On" : "Off"]</A><BR>"
+	dat += "Cat scream: <A href='?src=\ref[src];action=toggle;scream=cat'>[('sound/misc/cat.ogg' in sounds) ? "On" : "Off"]</A><BR>"
+	dat += "Lizard scream: <A href='?src=\ref[src];action=toggle;scream=lizard'>[('sound/misc/lizard.ogg' in sounds) ? "On" : "Off"]</A><BR>"
+	dat += "Bird scream: <A href='?src=\ref[src];action=toggle;scream=caw'>[('sound/misc/caw.ogg' in sounds) ? "On" : "Off"]</A><BR>"
+	var/datum/browser/popup = new(user, "screambot", name, 300, 300)
+	popup.set_content(dat)
+	popup.open()
+	return
+
+/obj/machinery/bot/screambot/Topic(href, href_list)
+	if(..() || panel_open)
+		return
+
+	if(href_list["action"] == "toggle")
+		switch(href_list["scream"])
+			if("human")
```

Comment:
For all of the += can't you use |= instead? And I wonder can you do &= on lists to remove an object or all instances of an object from the list